src/cpp/r/RSexp.cpp

```diff
@@ -838,6 +854,121 @@ bool inherits(SEXP object, const char* S3Class)
    return Rf_inherits(object, S3Class);
 }
 
+std::set<std::string> makeNsePrimitives()
+{
+   std::set<std::string> nsePrimitives;
+   nsePrimitives.insert("quote");
+   nsePrimitives.insert("substitute");
+   nsePrimitives.insert("match.call");
+   nsePrimitives.insert("library");
+   nsePrimitives.insert("require");
+   nsePrimitives.insert("enquote");
+   nsePrimitives.insert("bquote");
+   nsePrimitives.insert("expression");
+   nsePrimitives.insert("evalq");
+   nsePrimitives.insert("subset");
+   return nsePrimitives;
+}
+
+const std::set<std::string>& nsePrimitives()
+{
+   static const std::set<std::string> set = makeNsePrimitives();
+   return set;
+}
+
+// Attempts to find calls to functions which perform NSE.
+bool maybePerformsNSEImpl(SEXP node,
```

Comment:
Sorry, I didn't mean it wasn't correct, I meant it wasn't _obviously_ correct, in that you really have to look at it to understand what's happening.